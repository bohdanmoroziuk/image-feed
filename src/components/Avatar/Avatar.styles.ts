import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  avatar: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  initials: {
    color: 'white',
  },
});
