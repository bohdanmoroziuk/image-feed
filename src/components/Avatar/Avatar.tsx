import { FunctionComponent, useMemo } from 'react';
import { View, Text, ColorValue } from 'react-native';

import styles from './Avatar.styles';

export interface AvatarProps {
  size: number;
  initials: string;
  color: ColorValue;
}

export const Avatar: FunctionComponent<AvatarProps> = ({
  size,
  initials,
  color,
}) => {
  const circleStyle = useMemo(() => ({
    width: size,
    height: size,
    borderRadius: size / 2,
    backgroundColor: color,
  }), [size, color]);

  return (
    <View style={[styles.avatar, circleStyle]}>
      <Text style={styles.initials}>
        {initials}
      </Text>
    </View>
  );
};
