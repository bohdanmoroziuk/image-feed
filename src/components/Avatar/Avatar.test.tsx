import { render } from '@testing-library/react-native';

import { Avatar } from './Avatar';

describe('Avatar', () => {
  it('renders successfully', () => {
    const { toJSON } = render(<Avatar color="teal" initials="FL" size={35} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
