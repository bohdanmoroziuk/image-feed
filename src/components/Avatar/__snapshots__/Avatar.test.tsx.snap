// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders successfully 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "teal",
        "borderRadius": 17.5,
        "height": 35,
        "width": 35,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    FL
  </Text>
</View>
`;
